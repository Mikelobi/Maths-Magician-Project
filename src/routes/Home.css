article {
  padding-top: 2em;
  width: 65%;
  margin: 0 auto;
}

h3 {
  font-size: 2em;
  text-align: center;
}

p {
  line-height: 1.6;
  letter-spacing: 0.03em;
  font-size: 1.1em;
}

span {
  color: rgb(223, 192, 57);
  font-weight: bold;
}

a {
  text-decoration: none;
  color: rgb(123, 230, 62);
  font-weight: bold;
}
