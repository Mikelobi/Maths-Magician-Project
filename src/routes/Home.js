import Navigation from '../components/Navigation';
import './Home.css';

export default function Home() {
  return (
    <section>
      <Navigation />
      <article>
        <h3>Welcome to your world of calculation</h3>
        <p>
          This Single Page Application have 3
          pages? Well that is the beauty of
          {' '}
          <span> React </span>
          {' '}
          using React router. Click the Navigations links to go to the
          {' '}
          <span> Calculator </span>
          {' '}
          page or click the
          {' '}
          <span> Quotes </span>
          {' '}
          link to checkout some lovely quotes that will inspire you in solving calculations.
        </p>
        <p>
          if you like the app, show your support by adding a star to the project
          on
          {' '}
          <a href="https://github.com/Mikelobi/Maths-Magician-Project">GitHub</a>
          {' '}
          .
        </p>
      </article>
    </section>
  );
}
