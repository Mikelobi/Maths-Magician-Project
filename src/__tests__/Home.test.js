import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../routes/Home';

describe('Test presence of Home component', () => {
  test('renders the Home component correctly', () => {
    const navComponent = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      )
      .toJSON();
    expect(navComponent).toMatchSnapshot();
  });
});
