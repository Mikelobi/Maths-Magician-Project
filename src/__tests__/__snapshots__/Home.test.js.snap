// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Home component renders the Home component correctly 1`] = `
<section>
  <div
    className="header"
  >
    <div>
      <h1
        className="header-title"
      >
        Math Magicians
      </h1>
    </div>
    <nav
      className="navigation"
    >
      <li
        className="navLink"
      >
        <a
          aria-current="page"
          className="link active"
          data-testid={1}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          className="link"
          data-testid={2}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          className="link"
          data-testid={3}
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </nav>
  </div>
  <article>
    <h3>
      Welcome to your world of calculation
    </h3>
    <p>
      This Single Page Application have 3 pages? Well that is the beauty of
       
      <span>
         React 
      </span>
       
      using React router. Click the Navigations links to go to the
       
      <span>
         Calculator 
      </span>
       
      page or click the
       
      <span>
         Quotes 
      </span>
       
      link to checkout some lovely quotes that will inspire you in solving calculations.
    </p>
    <p>
      if you like the app, show your support by adding a star to the project on
       
      <a
        href="https://github.com/Mikelobi/Maths-Magician-Project"
      >
        GitHub
      </a>
       
      .
    </p>
  </article>
</section>
`;
